### PR TITLE
Add 'expand all' checkbox on menu block config form.

### PR DESCRIPTION
**Pull request by opi**
Implements https://github.com/backdrop/backdrop-issues/issues/2773

### CODE REVIEW

**Comment by opi** [comment]
Feel free to write proper title & description for this checkbox, english is not my native language.

File: core/modules/system/system.menu.inc [hunk 1]
```diff
@@ -109,6 +110,17 @@ function system_menu_block_form($config) {
     '#default_value' => $config['toggle'],
     '#description' => t('A menu toggle button, commonly known as a hamburger icon, allows the menu to be hidden initially on devices with small screens, but then easily expanded.'),
   );
+  $form['expand_all'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Expand all'),
```
Comment (jenlampton) [inline core/modules/system/system.menu.inc]: Can we change this text to `Show all menu links`. Checkbox labels should be self-explanatory whenever possible.

File: core/modules/system/system.menu.inc [hunk 2]
```diff
@@ -109,6 +110,17 @@ function system_menu_block_form($config) {
...
+    '#default_value' => $config['expand_all'],
+    '#description' => t('Show every menu children, not only active trail.'),
```
Comment (jenlampton) [inline core/modules/system/system.menu.inc]: Can we change this text to `When checked, all menu links will be displayed. When unchecked, only menu links in the active trail will be displayed.` This text explains in more detail what happens both when checked and unchecked.

**Comment by opi** [comment]
thanks for this better wording

File: core/modules/system/system.menu.inc [hunk 3]
```diff
@@ -19,6 +19,7 @@ function system_menu_block_defaults($menu_name) {
     'level' => 1,
     'menu_name' => $menu_name,
     'style' => NULL,
+    'expand_all' => FALSE,
```
Comment (jenlampton) [inline core/modules/system/system.menu.inc:22]: nice catch 👍

**Comment by jenlampton** [comment]
Looks great!

**Comment by quicksketch** [comment]
Re-opening to get a new sandbox.

**Comment by quicksketch** [comment]
Merged into 1.x for 1.8.0.

**Pull request merged**